src/collection-view.js

```diff
@@ -446,45 +381,48 @@ const CollectionView = Backbone.View.extend({
     return this.collection.indexOf(view.model);
   },
 
-  // This method re-filters the children views and re-renders the results
+  // This method filters the children views and renders the results
   filter() {
     if (this._isDestroyed) { return this; }
 
-    if (!this.children.length) { return this; }
+    this._filterChildren();
 
-    const filteredViews = this._filterChildren();
-
-    this._renderChildren(filteredViews);
+    this._renderChildren();
 
     return this;
   },
 
   _filterChildren() {
-    const viewFilter = this._getFilter();
-    const addedViews = this._addedViews;
+    if (!this._children.length) { return; }
 
-    delete this._addedViews;
+    const viewFilter = this._getFilter();
 
     if (!viewFilter) {
-      if (addedViews) { return addedViews; }
+      const shouldReset = this.children.length !== this._children.length;
+
+      this.children._set(this._children._views, shouldReset);
```

Comment:
This will "reset" the children if the current `this.children` length doesn't match all of the views.  But if they do match, then we can assume the indexes are correct and just `_set` for a possible change to the array sort